File: week6/sentimental-readability/readability.py
from cs50 import get_string


def count_letters(text):
    alphabet = "abcdefghijklmnopqrstuvwxyz"
    letters = 0
    for char in text:
        if char.lower() in alphabet:
            letters += 1
    return letters


def count_words(text):
    words = 1
    for char in text:
        if char == " ":
            words += 1
    return words


def count_sentences(text):
    sentence_marker = ".!?"
    sentences = 0
    for char in text:
        if char in sentence_marker:
            sentences += 1
    return sentences
    

text = get_string("Text: ")
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

index = 0.0588 * (letters * 100 / words) - 0.296 * (sentences * 100 / words) - 15.8

if index < 1:
    print("Before Grade 1\n")
elif index >= 16:
    print("Grade 16+\n")
else:
    print(f"Grade {round(index)}")

File: week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int count_words(string text);
int count_letters(string text);
int count_sentences(string text);

int main(void)
{

    string text = get_string("Text: ");
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);
    double index = 0.0588 * (letters * 100 / words) - 0.296 * (sentences * 100 / words) - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)(index + 0.5)); //to round the number
    }
}


int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

